src/login/user-runtime-dir.c

```diff
@@ -148,6 +148,11 @@ int main(int argc, char *argv[]) {
 
         user = argv[2];
         r = get_user_creds(&user, &uid, &gid, NULL, NULL);
+        if (r == -ESRCH && streq(argv[1], "stop")) {
+                /* The user may be already removed. Fallback to parse_uid() if the requested operation is 'stop'. */
+                if (parse_uid(user, &uid) >= 0)
+                        r = 0;
+        }
```

Comment:
Hmm, I think this should work differently: in the stop case we should not consult the password database at first, i.e. directly go for parse_uid(), and use a fallback on get_user_creds() if that doesn't work...  Which means I think this code should actually be moved inside of the if branches below, and the start branch should use get_user_creds() as before, but  the stop branch should use parse_uid() first, with a fallback to get_user_creds(). This way the gid field doesn't have to be requested at all, which is pretty nice, too

It's generally a